src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ClickOnce.targets

```diff
@@ -0,0 +1,77 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ClickOnce.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <PublishDirName>app.publish</PublishDirName>
+    <PublishDir>$(OutputPath)app.publish\</PublishDir>
+    <GenerateManifests>true</GenerateManifests>
+    <_DeploymentSignClickOnceManifests Condition="'$(SignManifests)' == 'true'">true</_DeploymentSignClickOnceManifests>
+    <PublishProtocolProviderTargets>ClickOncePublish</PublishProtocolProviderTargets>
+  </PropertyGroup>
+
+  <!-- 
+    For .NET Core ClickOnce publish in Single File mode, the ClickOnce manifest generation needs to run 
+    after ComputeFilesToPublish. This is so that the bundle EXE which is included in the ClickOnce manifest 
+    is generated before the manifest is created.
+  -->
+  <PropertyGroup Condition="'$(PublishSingleFile)' == 'true'">
+    <DeploymentComputeClickOnceManifestInfoDependsOn >
+      $(DeploymentComputeClickOnceManifestInfoDependsOn);
+      ComputeFilesToPublish
+    </DeploymentComputeClickOnceManifestInfoDependsOn>
+  </PropertyGroup>
+
+  <!-- Ensure runtimeconfig.json is built before clickonce manifest generation -->
+  <PropertyGroup Condition="'$(GenerateRuntimeConfigurationFiles)'=='true'">
+    <GenerateManifestsDependsOn>
+       GenerateBuildRuntimeConfigurationFiles;
+       $(GenerateManifestsDependsOn)
+    </GenerateManifestsDependsOn>
+  </PropertyGroup>
+
+  <!-- Ensure deps.json is built before clickonce manifest generation -->
+  <PropertyGroup Condition="'$(GenerateDependencyFile)'=='true'">
+    <GenerateManifestsDependsOn>
+      GenerateBuildDependencyFile;
+      $(GenerateManifestsDependsOn)
+    </GenerateManifestsDependsOn>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(GenerateRuntimeConfigurationFiles)'=='true'">
+    <ProjectRuntimeConfigFilesForClickOnce Include="$(ProjectRuntimeConfigFilePath)"/>
+  </ItemGroup>
+  
+  <ItemGroup Condition="'$(GenerateDependencyFile)' == 'true'">
+    <ProjectDepsFilesForClickOnce Include="$(ProjectDepsFilePath)"/>
```

Comment:
Using ProjectDepsFilePath will be correct in the default new project scenario, but there are scenarios where a second publish specific deps file is created and you'll need to use that.  To reproduce this issue, create a new WPF .NET Core app, add a Nuget reference to "Microsoft.EntityFrameworkCore.Tools" and then build and run your app.  The app will likely fail to run, and in event viewer you'll see the reason is that there was a dependency that isn't found.

Fortunately there's already a property that tells you which deps file to use, and I just fixed this exact issue for wapproj so you can follow this example: https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/270214

Basically this will need to be changed to:

<ProjectDepsFilesForClickOnce Condition="'$(_UseBuildDependencyFile)' == 'true'" Include="$(ProjectDepsFilePath)"/>
<ProjectDepsFilesForClickOnce Condition="'$(_UseBuildDependencyFile)' != 'true'" Include="$(PublishDepsFilePath)"/>

And you may also need to depend on target _ComputeUseBuildDependencyFile

Reply:
It would be ideal if ClickOnce can reuse the fixes made for Installer Projects automatically, by consuming same output groups. That way a single fix would solve the publishing issue for all types. ClickOnce publishing is based on folder publishing, and I presume Installer Projects depend on the same.

Reply:
Updated ClickOnce targets to use correct deps json file based on Luke's comment.

Reply:
@NikolaMilosavljevic ClickOnce uses output groups that are visible in msbuild as well as Visual Studio for ClickOnce processing (e.g. Built, RefrenceCopyLocalPaths, Content etc). We have ClickOnce UI that also iterates through files in known output groups to show in the UI. Since we need to keep them in sync, it is risky to change the output groups we consume.